From a3914e468d3bb9ef70862a0d81cb4fbb69c934cd Mon Sep 17 00:00:00 2001
From: Joerg Bornemann <joerg.bornemann@qt.io>
Date: Mon, 9 Nov 2020 14:10:29 +0100
Subject: [PATCH 055/171] Fix pcre2 feature conditions

The pcre2 feature was turned off when the user passed -system-pcre to
configure. The reason were faulty expressions in the enable/disable
fields.

Fixes: QTBUG-88295
Change-Id: I4edee49a9c797797ef3b80f955b48b5492fb4fc7
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 8b086b49b48e7c69413ce08aa534378e54633fe4)
---
 src/corelib/configure.json | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/configure.json b/src/corelib/configure.json
index c5e0423273..9b5d19d41b 100644
--- a/src/corelib/configure.json
+++ b/src/corelib/configure.json
@@ -727,8 +727,8 @@
         },
         "pcre2": {
             "label": "PCRE2",
-            "disable": "input.pcre == 'no' || input.pcre == 'system'",
-            "enable": "input.pcre == 'qt'",
+            "disable": "input.pcre == 'no'",
+            "enable": "input.pcre == 'qt' || features.system-pcre2",
             "output": [ "privateConfig" ]
         },
         "system-pcre2": {
-- 
2.31.1

